apps/src/storage/dataBrowser/dataVisualizer/VisualizerModal.jsx

```diff
@@ -196,9 +198,39 @@ class VisualizerModal extends React.Component {
     return options.join(', ');
   }
 
+  countCells = memoize((records, tableColumns, chartType) => {
+    if (chartType !== ChartType.CROSS_TAB) {
+      return undefined;
+    } else {
+      return records.reduce((bucket, record) => {
+        tableColumns.forEach(column => {
+          bucket[column] ||= new Set();
+          bucket[column].add(record[column]);
+        });
+        return bucket;
+      }, new Set());
+    }
+  });
```

Comment:
This function will iterate over the cross product and count the number of values for all of the headers.

Reply:
Having a little trouble following this, is it producing a count of values? or a set of distinct values per header (which then is counted elsewhere)?

Reply:
_Technically_ it's producing a `Set`, but then we just read the `size` of the set to get the count of unique values. No further calculations required.